### PR TITLE
[MRG] GUI n_drive_cells widget

### LINKED ISSUE #850: GUI: Network configuration different after GUI initialization
The GUI does not reconstruct the same network connectivity and drives after reconstruction from widget values. 

__Test:__
1. Load test network configuration `hnn_core/tests/assets/jones2009_3x3_drives.json`. This config has all drive types (evoked, bursty, poisson) and tonic bias added. 
2. Run a simulation 
3. Compare the Network used to run the simulation (derived from GUI widget values) to a Network created by the API `read_network_connectivity` function.

__Problems:__
1. Number of connections is not the same. There are more bursty and poisson connections.
    - The GUI passes weights and constants dictionaries by cell types to the add_x_drive functions. Because there is an input box for each cell type, even weights with 0 are passed into the drive construction. This creates additional unnecessary null-weight connections.
    -  The GUI doesn't have an input widget for the n_drive_cells parameter. It uses the defaults 'n_cells' for evoked and poisson, and 1 for bursty. The synchronous inputs checkbox does send a n_drive_cells value to drive construction but otherwise the parameter is hidden from user input. But with the default bursty drives are always synchronous. 
2. The order of the drives is not the same. They are sorted alphabetically when read into the GUI. 
    - gid_ranges and pos_dict are not the same as a consequence of 1 and 2
3. Tonic bias is not in the GUI-derived Network 
    - This is because there isn't a way to load tonics into the GUI from file yet. 
4. Cell parameters are not the same. This is being addressed in #848  

__To Do:__
- [x]  Fix drive order when loading drives https://github.com/jonescompneurolab/hnn-core/pull/851
- [x]  Add tonic loading from file #852 
- [x]  Filter weights and constants for only cells with non-zero weights during drive construction. #853
- [x]  Add n_drive_cells input widget #861

### PR DESCRIPTION
**Pull request by gtdang**
**Needs #853 merged first**
Commits for this work start at [4789e9e](https://github.com/jonescompneurolab/hnn-core/pull/861/commits/4789e9e6c7d475749f2c0cf4c010a7e0a2aaae25). Previous are in #853. 

Adds "number of drive cells" to the drive widgets. 

Screenshot 2024-08-14 at 10 06 04 AM

Work done:
1. Created widget for n_drive_cells parameter
2. Changed the Synchronous inputs checkbox to Cell Specific to align with the API keyword 
3. Added observer to disable the n_dive_cells widget if the cell specific checkbox is checked 

closes #850

### CODE REVIEW

File: hnn_core/gui/gui.py [hunk 1]
```diff
@@ -1048,21 +1048,32 @@ def _get_cell_specific_widgets(layout, style, location, data=None):
...
+                         weights_nmda=None, delays=None,
+                         n_drive_cells=None, cell_specific=None
+                         ):
     default_data = {
         'tstart': 0.,
         'tstart_std': 0.,
         'tstop': tstop_widget.value,
         'burst_rate': 7.5,
         'burst_std': 0,
-        'repeats': 1,
```
Comment (ntolley) [inline hnn_core/gui/gui.py:1061]: any reason to remove `repeats`?

File: hnn_core/gui/gui.py [hunk 2]
```diff
@@ -1048,21 +1048,32 @@ def _get_cell_specific_widgets(layout, style, location, data=None):
...
+        'numspikes': 1,
+        'n_drive_cells': 1,
+        'cell_specific': False,
         'seedcore': 14,
     }
-    if isinstance(data, dict):
-        default_data = _update_nested_dict(default_data, data)
+    data.update({'n_drive_cells': n_drive_cells,
```
Comment (ntolley) [inline hnn_core/gui/gui.py:1074]: Why not define these directly in `default_data`?

File: hnn_core/gui/gui.py (hunk 1 above)
Comment (gtdang) [reply hnn_core/gui/gui.py:1061]: I changed this to 'numspikes' to match the api for drive creation.

File: hnn_core/tests/test_gui.py [hunk 3]
```diff
@@ -551,22 +551,58 @@ def test_gui_synchronous_inputs(setup_gui):
     """Test if the GUI creates plot using synchronous_inputs."""
     gui = setup_gui
 
+    # Set cell_specific to False
+    gui.drive_widgets[0]['is_cell_specific'].value = False
+    # Check that the n_drive_cells is not disabled
+    assert not gui.drive_widgets[0]['n_drive_cells'].disabled
+
+    # set synch inputs to first driver in simulation
```
Comment (ntolley) [inline hnn_core/tests/test_gui.py:559]:
```suggestion
    # set sync inputs to first driver in simulation
```

**Pull request merged**